js/reducers/navigation.js

```diff
@@ -4,22 +4,26 @@ import { type NavigationState, NavigationActions } from 'react-navigation'
 
 import { type Action } from '../actions/types'
 import { USER_LOGGED_IN_ACTION, USER_LOGGED_OUT_ACTION } from '../actions/login'
+import { LOGIN_SUCCESS } from '../store/actions/constants'
 import AppNavigator from '../navigation/AppNavigator'
-
 const INITIAL_STATE = AppNavigator.router.getInitialState()
 
 const reducer = (
   state: NavigationState = INITIAL_STATE,
   action: Action
 ): NavigationState => {
+  let nextState: ?NavigationState
   switch (action.type) {
     case USER_LOGGED_IN_ACTION:
+    case LOGIN_SUCCESS:
       // On user login return to the IngressScreen
-      return AppNavigator.router.getInitialState()
+      nextState = AppNavigator.router.getInitialState()
```

Comment:
I'd just return  AppNavigator.router.getInitialState() with a default case for state (so you dno't need to mutate nextState)

Reply:
I actually would like to avoid using the `default` case and make all the cases explicit